graphql_compiler/tests/snapshot_tests/test_graphql_schema_from_schema_data.py

```diff
@@ -0,0 +1,88 @@
+from ...schema_generation.schema_properties import (
+    ORIENTDB_BASE_EDGE_CLASS_NAME, ORIENTDB_BASE_VERTEX_CLASS_NAME, PROPERTY_TYPE_DECIMAL_ID,
+    PROPERTY_TYPE_EMBEDDED_SET_ID, PROPERTY_TYPE_LINK_ID, PROPERTY_TYPE_STRING_ID
+)
+
+
+class GraphQLSchemaGenerationTests(TestCase):
+
+    def test_graphql_schema_generation_from_schema_data_api(self):
+        orientdb_schema_data = [
+            {
+                'name': ORIENTDB_BASE_VERTEX_CLASS_NAME,
+                'abstract': False,
+                'properties': []
+            },
+            {
+                'name': ORIENTDB_BASE_EDGE_CLASS_NAME,
+                'abstract': False,
+                'properties': []
+            },
+            {
+                'name': 'Entity',
+                'abstract': True,
+                'superClasses': [ORIENTDB_BASE_VERTEX_CLASS_NAME],
+                'properties': [
+                    {
+                        'name': 'name',
+                        'type': PROPERTY_TYPE_STRING_ID,
+                    }
+                ]
+            },
+            {
+                'name': 'Person',
+                'abstract': False,
+                'superClasses': ['Entity'],
+                'properties': [
+                    {
+                        'name': 'net_worth',
+                        'type': PROPERTY_TYPE_DECIMAL_ID,
+                    },
+                    {
+                        'name': 'alias',
+                        'type': PROPERTY_TYPE_EMBEDDED_SET_ID,
+                        'linkedType': PROPERTY_TYPE_STRING_ID,
+                        'defaultValue': '{}'
+                    }
+                ],
+            },
+            {
+                'name': 'Person_LivesIn',
+                'abstract': False,
+                'customFields': {
+                    'human_name_in': 'Person',
+                    'human_name_out': 'Location where person lives',
+                },
+                'properties': [
+                    {
+                        'name': 'in',
+                        'type': PROPERTY_TYPE_LINK_ID,
+                        'linkedClass': 'Location',
+                    },
+                    {
+                        'name': 'out',
+                        'type': PROPERTY_TYPE_LINK_ID,
+                        'linkedClass': 'Person',
+                    }
+                ],
+                'superClass': ORIENTDB_BASE_EDGE_CLASS_NAME
+            },
+            {
+                'name': 'Location',
+                'abstract': False,
+                'superClasses': ['Entity'],
+                'properties': [
+                    {
+                        'name': 'description',
+                        'type': PROPERTY_TYPE_STRING_ID,
+                    }
+                ]
+            },
+        ]
+        schema, _ = get_graphql_schema_from_orientdb_schema_data(orientdb_schema_data)
+        self.assertMatchSnapshot(print_schema(schema))
```

Comment:
Do we actually need this to be a snapshot test here? This feels like a unit test to me since we have well-defined input and output, and a single function call being tested. Invoking all the snapshot testing machinery feels like overkill for this use case.

Reply:
I think it's a good idea because every time I've made a small change to the way we generate the schema, I've had to print it and then copy paste it into test_helpers. If I introduce this test as an unit test, every time we change the schema we now have to change two things.